Client/Frontend/Browser/BrowserViewController/BrowserViewController+FindInPageDelegate.swift

```diff
@@ -181,3 +181,101 @@ class WKWebViewFindStringFindDelegate: NSObject {
+    find(text, function: "find")
+  }
+
+  func findInPage(_ findInPage: FindInPageBar, didFindNextWithText text: String) {
+    findInPageBar?.endEditing(true)
+    find(text, function: "findNext")
+  }
+
+  func findInPage(_ findInPage: FindInPageBar, didFindPreviousWithText text: String) {
+    findInPageBar?.endEditing(true)
+    find(text, function: "findPrevious")
+  }
+
+  func findInPageDidPressClose(_ findInPage: FindInPageBar) {
+    updateFindInPageVisibility(visible: false)
+  }
+
+  func findInPageHelper(_ findInPageHelper: FindInPageScriptHandler, didUpdateCurrentResult currentResult: Int) {
+    findInPageBar?.currentResult = currentResult
+  }
+
+  func findInPageHelper(_ findInPageHelper: FindInPageScriptHandler, didUpdateTotalResults totalResults: Int) {
+    findInPageBar?.totalResults = totalResults
+  }
+
+  func findTextInPage(_ direction: TextSearchDirection) {
+    guard let seachText = findInPageBar?.text else { return }
+
+    find(seachText, function: direction.rawValue)
+  }
+  
+  func find(_ text: String, function: String) {
+    guard let webView = tabManager.selectedTab?.webView else { return }
+
+    if let delegate = webView.findInPageDelegate {
+      let backwards = function == TextSearchDirection.previous.rawValue
+
+      delegate.find(string: text, backwards: backwards) { [weak self] index, total in
+        guard let self = self else { return }
+        self.findInPageBar?.totalResults = Int(total)
+        self.findInPageBar?.currentResult = index
+      }
+    } else {
+      webView.evaluateSafeJavaScript(functionName: "__firefox__.\(function)", args: [text], contentWorld: FindInPageScriptHandler.scriptSandbox)
+    }
+  }
+  
+  func updateFindInPageVisibility(visible: Bool, tab: Tab? = nil) {
+    if visible {
+      if findInPageBar == nil {
+        let findInPageBar = FindInPageBar()
+        self.findInPageBar = findInPageBar
+        findInPageBar.delegate = self
+        
+        displayPageZoom(visible: false)
+        alertStackView.addArrangedSubview(findInPageBar)
+
+        findInPageBar.snp.makeConstraints { make in
+          make.height.equalTo(UIConstants.toolbarHeight)
+          make.edges.equalTo(alertStackView)
+        }
+
+        updateViewConstraints()
+
+        // We make the find-in-page bar the first responder below, causing the keyboard delegates
+        // to fire. This, in turn, will animate the Find in Page container since we use the same
+        // delegate to slide the bar up and down with the keyboard. We don't want to animate the
+        // constraints added above, however, so force a layout now to prevent these constraints
+        // from being lumped in with the keyboard animation.
+        findInPageBar.layoutIfNeeded()
+      }
+
+      self.findInPageBar?.becomeFirstResponder()
+    } else {
+      // Empty string should be executed in order to remove the text highlights before find script done
+      find("", function: "find")
```

Comment:
Before de-activating the Find-In Page to remove the highlights from the page so normal interaction can be done.